### PR TITLE
Add ZeroAsciiIgnoreCaseTrieCursor

### PR DESCRIPTION
**Pull request by sffc**
Last piece of the ZeroTrie improvements required to continue work on #4548 / #4031

I filed #4561 for additional follow-up work to try to reduce code duplication. I would like to not block this PR on code duplication concerns.

### CODE REVIEW

File: experimental/zerotrie/src/reader.rs [hunk 1]
```diff
@@ -410,25 +410,51 @@ pub(crate) fn get_parameterized<T: ZeroTrieWithOptions + ?Sized>(
...
 /// The input-output argument `trie` starts at the original trie and ends pointing to
 /// the sub-trie reachable by `c`.
-pub(crate) fn step_ascii_bsearch_only(trie: &mut &[u8], c: u8) {
+#[inline]
+pub(crate) fn step_parameterized<T: ZeroTrieWithOptions + ?Sized>(
+    trie: &mut &[u8],
+    c: u8,
+) -> Option<u8> {
+    // BinarySpans is tricky to implement because the state can no longer be simply a trie
+    debug_assert!(
```
Comment (robertbastian) [inline experimental/zerotrie/src/reader.rs:419]: Should this return `None` here without assertions?

File: experimental/zerotrie/src/cursor.rs [hunk 2]
```diff
@@ -87,6 +135,15 @@ pub struct ZeroTrieSimpleAsciiCursor<'a> {
     trie: ZeroTrieSimpleAscii<&'a [u8]>,
 }
 
+/// A cursor into a [`ZeroAsciiIgnoreCaseTrie`], useful for stepwise lookup.
+///
+/// For examples, see [`ZeroAsciiIgnoreCaseTrie::cursor()`].
+// Clone but not Copy: <https://stackoverflow.com/q/32324251/1407170>
+#[derive(Debug, Clone)]
+pub struct ZeroAsciiIgnoreCaseTrieCursor<'a> {
```
Comment (robertbastian) [inline experimental/zerotrie/src/cursor.rs:143]: not a big fan of adding yet another type, but you already know that

File: experimental/zerotrie/src/reader.rs (hunk 1 above)
Comment (sffc) [reply experimental/zerotrie/src/reader.rs:419]: Without assertions it returns None on line 468, if it actually hits a span node.

File: experimental/zerotrie/src/cursor.rs (hunk 2 above)
Comment (sffc) [reply experimental/zerotrie/src/cursor.rs:143]: I wrote some ideas on how to resolve this in https://github.com/unicode-org/icu4x/issues/4561

File: experimental/zerotrie/src/reader.rs (hunk 1 above)
Comment (robertbastian) [reply experimental/zerotrie/src/reader.rs:419]: There's already a debug assert there, why this as well? There should at least be a comment linking to the line and what the gigo value is.

Comment (sffc) [reply experimental/zerotrie/src/reader.rs:421]: I expanded these comments

**Review by robertbastian — APPROVED** [review]

**Pull request merged**